src/components/common/messaging/MessageBox.tsx

```diff
@@ -198,37 +201,63 @@ export default observer(({ channel }: Props) => {
         stopTyping();
         setMessage();
         setReplies([]);
-        playSound("outbound");
-
         const nonce = ulid();
-        dispatch({
-            type: "QUEUE_ADD",
-            nonce,
-            channel: channel._id,
-            message: {
-                _id: nonce,
-                channel: channel._id,
-                author: client.user!._id,
-
-                content,
-                replies,
-            },
-        });
 
-        defer(() => renderer.jumpToBottom(SMOOTH_SCROLL_ON_RECEIVE));
+        // sed style message editing.
+        // If the user types for example `s/abc/def`, the string "abc"
+        // will be replaced with "def" in their last sent message.
+        if (SED_REGEX.test(content)) {
+            renderer.messages.reverse();
+            const msg = renderer.messages.find(
+                (msg) => msg.author_id === client.user!._id,
+            );
+            renderer.messages.reverse();
+
+            if (msg) {
+                const [_, toReplace, newText, flags] =
+                    content.split(/(?<!\\)\//);
```

Comment:
Turns out [Safari doesn't like this](https://caniuse.com/js-regexp-lookbehind)

Reply:
rip

Reply:
only fixed **_this april_**